### PR TITLE
Added updates to URL Scan

### PR DESCRIPTION
**Pull request by penwoodjon**
- Added Referer Url from url response to the Overlay
- Added the Screenshot URL to the overlay
- Added the ability to submit a domain to urlscan, if you have an api key and have the option checked.

### CODE REVIEW

File: config/config.js [hunk 1]
```diff
@@ -95,11 +95,30 @@ module.exports = {
...
+      type: 'boolean',
+      userCanEdit: true,
+      adminOnly: false
+    },
+    {
+      key: 'submitUrl',
+      name: 'Submit URL for Scanning',
+      description:
+        'If there are no search results, this allows you to submit the url for scanning when searching On Demand',
+      default: true,
```
[the remark this answers is not in the document]
Comment (penwoodjon) [reply config/config.js:121]: Done!

File: integration.js [hunk 2]
```diff
@@ -322,13 +363,76 @@ function _isEntityBlacklisted(entity, options) {
...
+            url: body.url
+          }
+        }]
+      });
+      
+    }
+  });
+};
+
+const onDetails = (lookupObject, options, cb) => {
```
[the remark this answers is not in the document]
Comment (penwoodjon) [reply integration.js:418]: Sorry about that.  Accidentally committed something when I was messing around with some of the onDetails stuff we were discussing previously.  Just removed it.

File: integration.js [hunk 3]
```diff
@@ -141,11 +141,27 @@ function doLookup(entities, options, cb) {
...
+        result.entity.requestContext.requestType === 'OnDemand' &&
+        result.entity.isDomain &&
+        result.body &&
+        result.body.results &&
+        result.body.results.length === 0;
+
+      if (canSubmitUrl) {
+        lookupResults.push({
+          entity: result.entity,
+          isVolitile: true,
```
[the remark this answers is not in the document]
Comment (penwoodjon) [reply integration.js:158]: Good catch! Fixed!

File: templates/us-block.hbs [hunk 4]
```diff
@@ -1,163 +1,240 @@
...
-      {{#if result.task}}
-      <h1 class="p-title">
-      {{fa-icon "scanner" fixedWidth=true}}  Scan Information
-      </h1>
-      {{/if}}
+            {{#if result._id}}
+                <div>
+                    <span class="p-key">Screenshot URL: </span>
+                    <span class="p-value">
+                        https://urlscan.io/screenshots/{{result._id}}.png
```
[the remark this answers is not in the document]
Comment (penwoodjon) [reply templates/us-block.hbs:75]: Done!

File: templates/us-block.hbs [hunk 5]
```diff
@@ -1,163 +1,240 @@
...
-      {{#if result.page.asn}}
-      <div>
-          <span class="p-key">ASN: </span>
-          <span class="p-value">{{result.page.asn}}</span>
-      </div>
-      {{/if}}
-    </div>
-{{/each}}
+    {{#if (and details.refererLinks details.refererLinks.length)}}
+        <div class="p-title">Referred To URLs ({{details.refererLinks.length}}):</div>
```
[the remark this answers is not in the document]
Comment (penwoodjon) [reply templates/us-block.hbs:233]: Good catch! Fixed!

File: components/us-block.js [hunk 6]
```diff
@@ -1,4 +1,43 @@
-'use strict';
 polarity.export = PolarityComponent.extend({
-  details: Ember.computed.alias('block.data.details')
+  details: Ember.computed.alias('block.data.details'),
+  showDetails: false,
```
[the remark this answers is not in the document]
Comment (penwoodjon) [reply components/us-block.js:3]: You are correct.  Just removed it!

File: config/config.js [hunk 7]
```diff
@@ -23,7 +23,7 @@ module.exports = {
    * @optional
    */
   description: 'Searches the urlscan.io API and returns results from the most recent, relevant scan',
-  entityTypes: ['IPv4', 'IPv6', 'IPv4CIDR', 'domain', 'sha256'],
+  entityTypes: ['IPv4', 'IPv6', 'IPv4CIDR', 'domain', 'url', 'sha256'],
```
[the remark this answers is not in the document]
Comment (penwoodjon) [reply config/config.js:26]: Done!

**Comment by sarus** [comment]
@penwoodjon A couple of other issues.

1. We should not send any private IPs (RFC 1918) addresses for lookup.  You can test for this with something like:

```
if(entity.isIP && entity.isPrivateIP){
    // skip this entity because it's a private IP
}
```
2. When I sent a private IP (in this case `192.168.1.1`), I got the following back in the overlay window which doesn't seem right:

image

The data coming back is:

```
       "data": {
          "summary": [],
          "details": {
            "results": [],
            "total": 0
          }
        }
```

I feel like in this scenario this should be considered a miss.

**Comment by penwoodjon** [comment]
> @penwoodjon I think we need a bit more explanation around submitting your own domain or URL. For example, after I submit, I click on the screenshot URL and the `View Details` URL and both will go to a 404 not found page. Is this expected behavior? If it is, we need to explain to the user what to expect. For example, is it a 404 because the submitted URL is still being scanned by urlscan? Or is it a 404 because the URL was not found or is invalid.

Makes sense.  I've added a note when you get the submission results.  I've noticed the 404 with the `View Details` link is temporary, but if the url is invalid the screenshot url won't ever resolve.  Not sure how best to communicate that, but can always update the note however you think is best.

**Pull request merged**